qadence/backends/pyqtorch/convert_ops.py

```diff
@@ -76,24 +76,32 @@ def convert_block(
                 # TODO: use the hevo_algo configuration here to switch between different algorithms
                 # for executing the Hamiltonian evolution
                 operation=pyq.HamiltonianEvolution(
-                    block.qubit_support,
+                    list(block.qubit_support),
                     n_qubits,
-                    n_steps=config.n_steps_hevo,
                 ),
                 block=block,
                 config=config,
             )
+        elif block.n_supports == 1:
+            op = getattr(pyq, block.name)(block.qubit_support[0], config.get_param_name(block)[0])
         else:
-            op = ParametricPyQOperation(n_qubits, block, config)
+            op = getattr(pyq, block.name)(
+                block.qubit_support[0], block.qubit_support[1], config.get_param_name(block)[0]
+            )
```

Comment:
This does not look a generic condition. Maybe you can replace it with: 

```python
op = getattr(pyq, block.name)(*block.qubit_support, config.get_param_name(block)[0])
```

Reply:
check now